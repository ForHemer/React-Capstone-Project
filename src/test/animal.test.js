import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Animal from '../components/animal/animal';

describe('Testing Animal', () => {
  it('Render the Details page, should match to the snapshot', () => {
    const list = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Animal />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(list).toMatchSnapshot();
  });
});
