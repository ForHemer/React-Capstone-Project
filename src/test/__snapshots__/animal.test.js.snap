// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Animal Render the Details page, should match to the snapshot 1`] = `
<div
  className="animalWidth"
>
  <a
    href="/AnimalDetail/undefined"
    onClick={[Function]}
  >
    <button
      className="buttonRight"
      onClick={[Function]}
      type="button"
    >
      DETAIL
    </button>
  </a>
  <ul
    className="listStyle"
  >
    <li
      style={
        Object {
          "paddingBottom": "2rem",
        }
      }
    >
      <img
        className="card-img"
        style={
          Object {
            "height": "120px",
            "width": "120px",
          }
        }
      />
    </li>
    <li
      className="listInfo"
    >
      <span>
        Name :
      </span>
      <span />
    </li>
    <li
      className="listInfo"
    >
      <span>
        Type :
      </span>
      <span />
    </li>
    <li
      className="listInfo"
    >
      <span>
        Lifespan :
      </span>
      <span />
    </li>
    <li
      className="listInfo"
    >
      <span>
        Weight (max) :
      </span>
      <span />
    </li>
  </ul>
</div>
`;
